src/index.js

```diff
@@ -1,22 +1,22 @@
 // @flow
-type EqualityFn = (a: any, b: any) => boolean;
+type EqualityFn = (a: mixed, b: mixed) => boolean;
 
-const simpleIsEqual = (a: any, b: any): boolean => a === b;
+const simpleIsEqual: EqualityFn = (a: mixed, b: mixed): boolean => a === b;
 
-export default function (resultFn: Function, isEqual?: EqualityFn = simpleIsEqual) {
-  let lastThis: any;
-  let lastArgs: Array<any> = [];
-  let lastResult: any;
+export default function <ResultFn: (...Array<any>) => mixed>(resultFn: ResultFn, isEqual?: EqualityFn = simpleIsEqual): ResultFn {
```

Comment:
Yup, that's right. While you're not enforcing inputs, you can at least be sure that you aren't handling the output incorrectly.